### PR TITLE
Add bitwise BitVec <-> Float conversion

### PR DESCRIPTION
**Pull request by N1ark**
Properly support converting bitvectors to/from floats, without any sort of rounding. This is non-trivial, because Smt-lib does not expose a `fp_to_bv` operator, because all NaN values are modelled as one "NaN", that can represent any of the NaN bit-patterns.

To convert a float `f` to a bv, we need to create a new nondet bitvector `b` and assume `to_fp b == f`.
```lisp
(declare-fun b () (_ BitVec m))
(assert (= ((_ to_fp eb sb) b) f))
```

Because of the nondeterministic handling of NaN, there is a caveat to this support: while floats remain NaN through transmutation, the bit-pattern is not preserved. This is not ideal but is good enough. Doing better would require much heavier engineering.

### CODE REVIEW

File: soteria-rust/test/cram/simple.t/run.t [hunk 1]
```diff
@@ -167,3 +167,21 @@ Check corner cases with permissive provenance, around transmutes
...
+Test transmutations keeping the bit-patterns the same
+  $ soteria-rust rustc transmute_roundtrip.rs --clean --no-timing
+  Compiling... done in <time>
+  note: one_way_u32_f32: done in <time>, ran 1 branch
+  PC 1: (bv2f[F32](V|1|) == bv2f[F32](V|2|)) /\ !(fis(NaN)(bv2f[F32](V|1|))) /\
+        (V|1| == V|2|)
+  
+  note: one_way_f32_u32: done in <time>, ran 2 branches
+  PC 1: (bv2f[F32](V|2|) == V|1|) /\ fis(NaN)(V|1|)
+  PC 2: (bv2f[F32](V|2|) == V|1|) /\ !(fis(NaN)(V|1|))
```
[the remark this answers is not in the document]
Comment (giltho) [reply soteria-rust/test/cram/simple.t/run.t]: Really nice!
